nexus/src/app/sagas/snapshot_create.rs

```diff
@@ -204,8 +239,32 @@ impl NexusSaga for SagaSnapshotCreate {
         // (DB) Creates a record of the snapshot, referencing both the
         // original disk ID and the destination volume
         builder.append(create_snapshot_record_action());
-        // (Sleds) Sends a request for the disk to create a ZFS snapshot
-        builder.append(send_snapshot_request_action());
+
+        if !params.use_the_pantry {
+            // (Sleds) If the disk is attached to an instance, send a
+            // snapshot request to sled-agent to create a ZFS snapshot.
+            builder.append(send_snapshot_request_to_sled_agent_action());
+        } else {
+            // (Pantry) If the disk is _not_ attached to an instance:
+            // "attach" the disk to the pantry
+            builder.append(attach_disk_to_pantry_action());
+
+            // (Pantry) Record the address of a Pantry service
+            builder.append(get_pantry_address_action());
+
+            // (Pantry) Call the Pantry's /attach
+            builder.append(call_pantry_attach_for_disk_action());
+
+            // (Pantry) Call the Pantry's /snapshot
+            builder.append(call_pantry_snapshot_for_disk_action());
+
+            // (Pantry) Call the Pantry's /detach
+            builder.append(call_pantry_detach_for_disk_action());
+
+            // (Pantry) Clear attach_instance_id ZZZ verify, maybe DB too?
```

Comment:
ZZZ?

Reply:
Yeah, more leftover stuff I forgot to remove, fixed in 2428c0494e5a53f8fbf751b90b15bae1094edbe4